portality/view/forms.py

```diff
@@ -56,13 +56,25 @@ class ArticleForm(Form):
 
     def __init__(self, *args, **kwargs):
         super(ArticleForm, self).__init__(*args, **kwargs)
+        if "id" in kwargs:
+            a = models.Article.pull(kwargs["id"])
+            self.id = kwargs["id"]
+            if ("method" not in kwargs or kwargs["method"] != "post"):
+                bibjson = a.bibjson()
+                article_form.ArticleFormXWalk.obj2form(self, bibjson)
```

Comment:
I'm not quite sure what's going on here, but I'm against embedding the form crosswalk into the WTForms instance, it seems the wrong place.  I think we should coalesce everything into a FormContext, in which case the crosswalks will sit comfortably there.